push-build.sh

```diff
@@ -96,7 +97,19 @@ RELEASE_BUCKET=${FLAGS_bucket:-"kubernetes-release-dev"}
 [[ $KUBE_GCS_UPDATE_LATEST == "n" ]] && FLAGS_noupdatelatest=1
 
 KUBECTL_OUTPUT=$(cluster/kubectl.sh version --client 2>&1 || true)
-if [[ "$KUBECTL_OUTPUT" =~ GitVersion:\"(${VER_REGEX[release]}(\.${VER_REGEX[build]})?)\", ]]; then
+if [[ "$KUBECTL_OUTPUT" =~ GitVersion:\"(${VER_REGEX[release]}(\.${VER_REGEX[build]})?-dirty)\", ]]; then
```

Comment:
I wonder if this should be changed from `-dirty` to `(-dirty)?` to eliminate duplication of the `GitVersion` regex, with a separate check for whether it's tagged `-dirty`, e.g. looking at `GitTreeState`.